File: wsTripXML/Code/OTAClasses/OTAAirPrice.vb

Option Strict Off
Option Explicit On 

Imports System.Xml.Serialization

Namespace wsTravelTalk.wmAirPrice

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class OriginDestinationOption

        '<remarks/>
        <System.Xml.Serialization.XmlElementAttribute("FlightSegment")> _
        Public FlightSegment() As FlightSegment
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class OriginDestinationOptions

        '<remarks/>
        <System.Xml.Serialization.XmlArrayItemAttribute(GetType(FlightSegment), IsNullable:=False)> _
        Public OriginDestinationOption()() As FlightSegment
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class AirItinerary

        '<remarks/>
        <System.Xml.Serialization.XmlArrayItemAttribute("OriginDestinationOption", IsNullable:=False), _
         System.Xml.Serialization.XmlArrayItemAttribute(IsNullable:=False, NestingLevel:=1)> _
        Public OriginDestinationOptions()() As FlightSegment

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public DirectionInd As AirItineraryDirectionInd

        '<remarks/>
        <System.Xml.Serialization.XmlIgnoreAttribute()> _
        Public DirectionIndSpecified As Boolean
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class FlightSegment

        '<remarks/>
        Public DepartureAirport As DepartureAirport

        '<remarks/>
        Public ArrivalAirport As ArrivalAirport

        '<remarks/>
        Public OperatingAirline As OperatingAirline

        '<remarks/>
        <System.Xml.Serialization.XmlElementAttribute("Equipment")> _
        Public Equipment() As Equipment

        '<remarks/>
        Public MarketingAirline As MarketingAirline

        '<remarks/>
        Public MarriageGrp As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public DepartureDateTime As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public ArrivalDateTime As String

        '<remarks/>
        <System.Xml.Serialization.XmlIgnoreAttribute()> _
        Public ArrivalDateTimeSpecified As Boolean

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public StopQuantity As Integer

        '<remarks/>
        <System.Xml.Serialization.XmlIgnoreAttribute()> _
        Public StopQuantitySpecified As Boolean

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public RPH As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public InfoSource As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public FlightNumber As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public TourOperatorFlightID As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public ResBookDesigCode As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public ActionCode As FlightSegmentActionCode

        '<remarks/>
        <System.Xml.Serialization.XmlIgnoreAttribute()> _
        Public ActionCodeSpecified As Boolean

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public NumberInParty As Integer

        '<remarks/>
        <System.Xml.Serialization.XmlIgnoreAttribute()> _
        Public NumberInPartySpecified As Boolean

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public FlightContext As String
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class DepartureAirport

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public LocationCode As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute(), _
         System.ComponentModel.DefaultValueAttribute("IATA")> _
        Public CodeContext As String = "IATA"

        '<remarks/>
        <System.Xml.Serialization.XmlTextAttribute()> _
        Public Value As String
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class ArrivalAirport

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public LocationCode As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute(), _
         System.ComponentModel.DefaultValueAttribute("IATA")> _
        Public CodeContext As String = "IATA"

        '<remarks/>
        <System.Xml.Serialization.XmlTextAttribute()> _
        Public Value As String
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class OperatingAirline

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public CompanyShortName As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public TravelSector As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public Code As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public CodeContext As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public FlightNumber As String

        '<remarks/>
        <System.Xml.Serialization.XmlTextAttribute()> _
        Public Value As String


    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class MarketingAirline

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public CompanyShortName As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public TravelSector As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public Code As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public CodeContext As String

        '<remarks/>
        <System.Xml.Serialization.XmlTextAttribute()> _
        Public Value As String
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class Equipment

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public AirEquipType As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute(), _
         System.ComponentModel.DefaultValueAttribute(False)> _
        Public ChangeofGauge As Boolean = False

        '<remarks/>
        <System.Xml.Serialization.XmlTextAttribute()> _
        Public Value As String
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class PassengerTypeQuantity

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public Age As Integer

        '<remarks/>
        <System.Xml.Serialization.XmlIgnoreAttribute()> _
        Public AgeSpecified As Boolean

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public Code As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public CodeContext As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute(DataType:="anyURI")> _
        Public URI As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public Quantity As Integer

        '<remarks/>
        <System.Xml.Serialization.XmlIgnoreAttribute()> _
        Public QuantitySpecified As Boolean

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute(), _
         System.ComponentModel.DefaultValueAttribute(True)> _
        Public Changeable As Boolean = True
    End Class

    '<remarks/>
    Public Enum FlightSegmentActionCode

        '<remarks/>
        OK

        '<remarks/>
        Waitlist

        '<remarks/>
        Other
    End Enum

    '<remarks/>
    Public Enum AirItineraryDirectionInd

        '<remarks/>
        OneWay

        '<remarks/>
        [Return]

        '<remarks/>
        Circle

        '<remarks/>
        OpenJaw

        '<remarks/>
        Other
    End Enum

End Namespace

File: src/wsTripXML/Code/OTAClasses/OTAAirPrice.cs

using System.Xml.Serialization;

namespace wsTripXML.wsTravelTalk.wmAirPrice
{

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class OriginDestinationOption
    {

        // <remarks/>
        [XmlElement("FlightSegment")]
        public FlightSegment[] FlightSegment;
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class OriginDestinationOptions
    {

        // <remarks/>
        [XmlArrayItem(typeof(FlightSegment), IsNullable = false)]
        public FlightSegment[][] OriginDestinationOption;
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class AirItinerary
    {

        // <remarks/>
        [XmlArrayItem("OriginDestinationOption", IsNullable = false)]
        [XmlArrayItem(IsNullable = false, NestingLevel = 1)]
        public FlightSegment[][] OriginDestinationOptions;

        // <remarks/>
        [XmlAttribute()]
        public AirItineraryDirectionInd DirectionInd;

        // <remarks/>
        [XmlIgnore()]
        public bool DirectionIndSpecified;
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class FlightSegment
    {

        // <remarks/>
        public DepartureAirport DepartureAirport;

        // <remarks/>
        public ArrivalAirport ArrivalAirport;

        // <remarks/>
        public OperatingAirline OperatingAirline;

        // <remarks/>
        [XmlElement("Equipment")]
        public Equipment[] Equipment;

        // <remarks/>
        public MarketingAirline MarketingAirline;

        // <remarks/>
        public string MarriageGrp;

        // <remarks/>
        [XmlAttribute()]
        public string DepartureDateTime;

        // <remarks/>
        [XmlAttribute()]
        public string ArrivalDateTime;

        // <remarks/>
        [XmlIgnore()]
        public bool ArrivalDateTimeSpecified;

        // <remarks/>
        [XmlAttribute()]
        public int StopQuantity;

        // <remarks/>
        [XmlIgnore()]
        public bool StopQuantitySpecified;

        // <remarks/>
        [XmlAttribute()]
        public string RPH;

        // <remarks/>
        [XmlAttribute()]
        public string InfoSource;

        // <remarks/>
        [XmlAttribute()]
        public string FlightNumber;

        // <remarks/>
        [XmlAttribute()]
        public string TourOperatorFlightID;

        // <remarks/>
        [XmlAttribute()]
        public string ResBookDesigCode;

        // <remarks/>
        [XmlAttribute()]
        public FlightSegmentActionCode ActionCode;

        // <remarks/>
        [XmlIgnore()]
        public bool ActionCodeSpecified;

        // <remarks/>
        [XmlAttribute()]
        public int NumberInParty;

        // <remarks/>
        [XmlIgnore()]
        public bool NumberInPartySpecified;

        // <remarks/>
        [XmlAttribute()]
        public string FlightContext;
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class DepartureAirport
    {

        // <remarks/>
        [XmlAttribute()]
        public string LocationCode;

        // <remarks/>
        [XmlAttribute()]
        [System.ComponentModel.DefaultValue("IATA")]
        public string CodeContext = "IATA";

        // <remarks/>
        [XmlText()]
        public string Value;
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class ArrivalAirport
    {

        // <remarks/>
        [XmlAttribute()]
        public string LocationCode;

        // <remarks/>
        [XmlAttribute()]
        [System.ComponentModel.DefaultValue("IATA")]
        public string CodeContext = "IATA";

        // <remarks/>
        [XmlText()]
        public string Value;
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class OperatingAirline
    {

        // <remarks/>
        [XmlAttribute()]
        public string CompanyShortName;

        // <remarks/>
        [XmlAttribute()]
        public string TravelSector;

        // <remarks/>
        [XmlAttribute()]
        public string Code;

        // <remarks/>
        [XmlAttribute()]
        public string CodeContext;

        // <remarks/>
        [XmlAttribute()]
        public string FlightNumber;

        // <remarks/>
        [XmlText()]
        public string Value;


    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class MarketingAirline
    {

        // <remarks/>
        [XmlAttribute()]
        public string CompanyShortName;

        // <remarks/>
        [XmlAttribute()]
        public string TravelSector;

        // <remarks/>
        [XmlAttribute()]
        public string Code;

        // <remarks/>
        [XmlAttribute()]
        public string CodeContext;

        // <remarks/>
        [XmlText()]
        public string Value;
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class Equipment
    {

        // <remarks/>
        [XmlAttribute()]
        public string AirEquipType;

        // <remarks/>
        [XmlAttribute()]
        [System.ComponentModel.DefaultValue(false)]
        public bool ChangeofGauge = false;

        // <remarks/>
        [XmlText()]
        public string Value;
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class PassengerTypeQuantity
    {

        // <remarks/>
        [XmlAttribute()]
        public int Age;

        // <remarks/>
        [XmlIgnore()]
        public bool AgeSpecified;

        // <remarks/>
        [XmlAttribute()]
        public string Code;

        // <remarks/>
        [XmlAttribute()]
        public string CodeContext;

        // <remarks/>
        [XmlAttribute(DataType = "anyURI")]
        public string URI;

        // <remarks/>
        [XmlAttribute()]
        public int Quantity;

        // <remarks/>
        [XmlIgnore()]
        public bool QuantitySpecified;

        // <remarks/>
        [XmlAttribute()]
        [System.ComponentModel.DefaultValue(true)]
        public bool Changeable = true;
    }

    // <remarks/>
    public enum FlightSegmentActionCode
    {

        // <remarks/>
        OK,

        // <remarks/>
        Waitlist,

        // <remarks/>
        Other
    }

    // <remarks/>
    public enum AirItineraryDirectionInd
    {

        // <remarks/>
        OneWay,

        // <remarks/>
        Return,

        // <remarks/>
        Circle,

        // <remarks/>
        OpenJaw,

        // <remarks/>
        Other
    }

}